src/sb_pointcloud_processing/src/DBSCAN.cpp

```diff
@@ -98,10 +109,92 @@ void DBSCAN::findNeighbors() {
         }
 
         // store the list of neighbours
-        this->_neighbors.insert({i, neighbors});
+        this->_neighbors[i] = neighbors;
+    }
+}
```

Comment:
Just out of curiosity, how does doing this with pthreads compare to using omp? Something like:
```
void DBSCAN::findNeighborsSequential() {
    #pragma omp parallel for 
    for (unsigned int i = 0; i < this->_pcl.size(); i++) {
        vector<unsigned int> neighbors;
        pcl::PointXYZ current_point = this->_pcl[i];

        // for current_point, determine all neighbour points that are within
        // predetermined radius
        for (unsigned int j = 0; j < this->_pcl.size(); j++) {
            if (i == j) continue;
            pcl::PointXYZ neighbor_point = this->_pcl[j];
            if (dist(current_point, neighbor_point) <= this->_radius) {
                neighbors.push_back(j);
            }
        }

        // store the list of neighbours
        this->_neighbors[i] = neighbors;
    }
}
```
If this works, I feel like it might be a bit cleaner
(for reference: https://bisqwit.iki.fi/story/howto/openmp/#ExampleInitializingATableInParallelMultipleThreads)

Reply:
confirmed that this works just fine